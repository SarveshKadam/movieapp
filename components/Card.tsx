import {
  Dimensions,
  ImageBackground,
  StyleSheet,
  Text,
  View
} from "react-native";

const windowWidth = Dimensions.get("window").width;
const numColumns = windowWidth >= 540 ? 3 : 2;
const itemWidth = windowWidth / numColumns - 2 * 16;
interface CardProps {
  title: string;
  vote_average: number;
  imageUri: string;
}

const PREFIX = "https://image.tmdb.org/t/p/w500/";

const Card: React.FC<CardProps> = ({ title, vote_average, imageUri }) => {
  return (
    <View style={styles.card}>
      <ImageBackground style={styles.image} source={{ uri: PREFIX + imageUri }}>
        <View style={styles.overlay}>
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.ratings}>‚≠ê {vote_average?.toFixed(1)}</Text>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    width: itemWidth,
    height: 250,
    backgroundColor: "white",
    marginVertical: 10,
    borderRadius: 5,
    overflow: "hidden",
    justifyContent: "center",
    alignItems: "center",
    elevation: 5,
    flex: 1 / numColumns,
  },
  image: {
    flex: 1,
    width: "100%",
    height: "100%",
    resizeMode: "cover",
  },
  overlay: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(0, 0, 0, 0.5)",
    justifyContent: "flex-end",
    padding: 10,
  },
  title: {
    fontSize: 14,
    color: "white",
    fontWeight: 600,
    marginBottom: 5,
  },
  ratings: {
    fontSize: 12,
    color: "white",
    fontWeight: 600,
  },
});

export default Card;
