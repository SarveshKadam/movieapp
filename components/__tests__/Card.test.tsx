import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Image } from 'react-native';
import Card from '../Card';
describe('Card Component', () => {
  const props = {
    title: 'Sample Title',
    vote_average: 7.58,
    imageUri: 'sample_image.jpg',
  };

  it('renders correctly with given props', () => {
    const tree = TestRenderer.create(<Card {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the title and release date correctly', () => {
    const tree = TestRenderer.create(<Card {...props} />);
    const instance = tree.root;

    const title = instance.findByProps({ children: props.title });
    const releaseDate = instance.findByProps({ children: props.vote_average });

    expect(title).toBeTruthy();
    expect(releaseDate).toBeTruthy();
  });

  it('renders the image with correct URI', () => {
    const tree = TestRenderer.create(<Card {...props} />);
    const instance = tree.root;

    const image = instance.findByType(Image);
    expect(image.props.source.uri).toBe('https://image.tmdb.org/t/p/w500/sample_image.jpg');
  });
});
