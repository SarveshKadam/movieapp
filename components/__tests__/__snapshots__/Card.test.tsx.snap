// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly with given props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "flex": 0.3333333333333333,
      "height": 250,
      "justifyContent": "center",
      "marginVertical": 10,
      "overflow": "hidden",
      "width": 218,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "height": "100%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://image.tmdb.org/t/p/w500/sample_image.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 14,
            "fontWeight": 600,
            "marginBottom": 5,
          }
        }
      >
        Sample Title
      </Text>
      <Text
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": 600,
          }
        }
      >
        ‚≠ê 
        7.6
      </Text>
    </View>
  </View>
</View>
`;
